kornia/contrib/sam/predictor.py

```diff
@@ -0,0 +1,122 @@
+        multimask_output: bool = True,
+        return_logits: bool = False,
+    ) -> SamPrediction:
+        """Predict masks for the given image based on the input prompts.
+
+        Args:
+            image: Batch of RGB image. Normally 8bits images (range of [0-255]), the model preprocess normalize the
+                   pixel values with the mean and std defined in its initialization. Expected to be into a float dtype.
+                   Shape :math:`(B, 3, H, W)` or :math:`(3, H, W)`.
+            point_coords: Point prompts to the model. Each point is in (X,Y) in pixels. Shape :math:`(B, N, 2)` or
+                          :math:`(N, 2)`. Where `N` is the number of points.
+            point_labels: Labels for the point prompts. 1 indicates a foreground point and 0 indicates a background
+                          point. Shape :math:`(B, N)` or :math:`(N)`. Where `N` is the number of points.
+            boxes: A box prompt to the model.
+            mask_input:  A low resolution mask input to the model, typically coming from a previous prediction
+                         iteration. Has shape :math:`(B, 1, H, W)` or :math:`(1, H, W)`, where for SAM, H=W=256.
+            multimask_output: If true, the model will return three masks. For ambiguous input prompts (such as a
+                              single click), this will often produce better masks than a single prediction. If only
+                              a single mask is needed, the model's predicted quality score can be used to select the
+                              best mask. For non-ambiguous prompts, such as multiple input prompts,
+                              multimask_output=False can give better results.
+            return_logits: If true, returns un-thresholded masks logits instead of a binary mask.
+
+        Returns:
+            A prediction with the masks, scores (IoU of each predicted mask), and the low resolution logits.
+        """
+        KORNIA_CHECK_SHAPE(
+            image, ['*', '3', 'H', 'W']
+        )  # FIXME: should be [('B', None), '3', 'H', 'W'], same for the others
+
+        dk = ['input']
+        _args: tuple[Any, ...] = (image.type(torch.float32),)
+
+        if isinstance(point_coords, (Keypoints, Tensor)) and isinstance(point_labels, Tensor):
+            KORNIA_CHECK_SHAPE(point_coords.data, ['*', 'N', '2'])
+            KORNIA_CHECK_SHAPE(point_labels.data, ['*', 'N'])
+            dk += ['keypoints']
+            _args += (point_coords,)
+
+        if isinstance(boxes, Boxes):
+            if boxes.mode == 'xyxy':
+                boxes_xyxy = boxes
+            else:
+                boxes_xyxy = Boxes(boxes.to_tensor(mode='xyxy'), mode='xyxy')
+
+            dk += ['bbox_xyxy']
+            _args += (boxes_xyxy,)
+
+        if isinstance(mask_input, Tensor):
+            KORNIA_CHECK_SHAPE(mask_input, ['*', '1', '256', '256'])
+
+        # FIXME: the augmentation sequential isn't accepting the k.geometry.Keypoints
```

Comment:
```python
sam_model = sam.load(model_type, checkpoint_path, device)
image = load_image(image_path, ImageLoadType.RGB8, device).float()[None, ...]
predictor = sam.SamPredictor(sam_model)
input_point = Keypoints(torch.tensor([[[500, 375]]], device=device, dtype=torch.float32)) # BxNx2
input_label = torch.tensor([[1]], device=device) # BxN
prediction = predictor(
    image=image,
    point_coords=input_point,
    point_labels=input_label,
    multimask_output=True,
)

```

This raises something internal on `AugmentationSequential`:
```output
---> 21     raise TypeError(f"Input keypoints is not a Tensor. Got: {type(keypoints)}.")

TypeError: Input keypoints is not a Tensor. Got: .
```

@shijianjian can help with this?

Reply:
Is raw tensor working? Or we cannot use keypoint at all? I will check it probably tomorrow.

Reply:
it's works with raw tensor